src/main/java/com/github/manolo8/darkbot/core/utils/EntityList.java

```diff
@@ -161,22 +177,18 @@ public void clear() {
             obstacles.clear();
             fakeNpc.removed();
 
-            allEntities.forEach(entities -> {
-                entities.forEach(Entity::removed);
+            for (Entity entity : all) {
+                entity.removed();
+                eventBroker.sendEvent(new EntityRemoveEvent(entity));
```

Comment:
Although probably irrelevant, the behavior here is different than the previous, in the earlier EntityRemoveEvent the entity has already been removed from the list (by the iterator), while in this one it hasn't yet been removed.

Would that potentially cause any issue?

Reply:
Entity was removed afterwards too in previous clear() method.
First was entity set to be removed -> send the event -> clear whole list -> repeat